packages/react/react/src/hooks/useImageLazyLoading.ts

```diff
@@ -0,0 +1,70 @@
+import { useRefEffect } from './useRefEffect';
+
+interface Props {
+  src: string;
+  threshold?: number | number[];
+  root?: Document | Element | null;
+  rootMargin?: string;
+  onAction?: () => void;
+}
+
+/** @tossdocs-ignore */
+export function useImageLazyLoading<Element extends HTMLImageElement>({
+  src,
+  rootMargin,
+  threshold,
+  root,
+  onAction,
+}: Props) {
+  const ref = useRefEffect<Element>(element => {
+    if (typeof IntersectionObserver === 'undefined') {
+      return;
+    }
+
+    if (element.tagName !== 'IMG') {
+      throw new Error("The target element can only be an 'img' tag.");
+    }
+
+    // Checks if undefined or null.
+    if (src == null) {
+      throw new Error('The useImageLazyLoading hook requires the src property as a required value.');
+    }
```

Comment:
These are not necessary because I think we only need to consider Typescript developers.